### PR TITLE
Fix autocompletion for paths with period 

### PR DESCRIPTION
**Pull request by DylanBulfin**
Fix for https://github.com/helix-editor/helix/issues/5136

Bug has to do with how the `file_name` method deals with paths ending in `.` or `/.`. I hardcoded the value for `file_name` for paths ending in `/.` and paths that only consist of a single period, and updated the logic for `path`.

### CODE REVIEW

File: helix-term/src/ui/mod.rs [hunk 1]
```diff
@@ -468,14 +468,23 @@ pub mod completers {
...
-            let file_name = path
-                .file_name()
-                .and_then(|file| file.to_str().map(|path| path.to_owned()));
-
-            let path = match path.parent() {
-                Some(path) if !path.as_os_str().is_empty() => path.to_path_buf(),
-                // Path::new("h")'s parent is Some("")...
-                _ => std::env::current_dir().expect("couldn't determine current directory"),
+            let is_period = (input.ends_with("/.") && input.len() > 2)
+                || (input.starts_with('.') && input.len() == 1);
```
Comment (the-mikedavis) [inline helix-term/src/ui/mod.rs:472]: The second check `input.starts_with('.') && input.len() == 1` could be `input == "."`. The `is_tilde` check on L465 could be refactored similarly

File: helix-term/src/ui/mod.rs [hunk 2]
```diff
@@ -462,20 +462,28 @@ pub mod completers {
...
         } else {
-            let file_name = path
-                .file_name()
-                .and_then(|file| file.to_str().map(|path| path.to_owned()));
-
-            let path = match path.parent() {
-                Some(path) if !path.as_os_str().is_empty() => path.to_path_buf(),
-                // Path::new("h")'s parent is Some("")...
-                _ => std::env::current_dir().expect("couldn't determine current directory"),
+            let is_period = (input.ends_with("/.") && input.len() > 2) || input == ".";
```
Comment (dead10ck) [inline helix-term/src/ui/mod.rs:471]: Should this use `std::path::MAIN_SEPARATOR` instead of hardcoding `/`?

Comment (pascalkuthe) [reply helix-term/src/ui/mod.rs:471]: It should probably work for both seperators. There are many file formats that always / and convert to \ on windows (or don't you can use / aswell on windows these days AFAIK)

**Review by dead10ck — APPROVED** [review]
Ok this seems fine to me :+1:

**Pull request merged**